controller/configentry_controller_test.go

```diff
@@ -1843,292 +1843,36 @@ func TestConfigEntryControllers_doesNotCreateUnownedConfigEntry(t *testing.T) {
 	kubeNS := "default"
 
 	cases := []struct {
-		kubeKind            string
```

Comment:
I removed the cases for each config entry and instead added cases based on the value of the `source-datacenter` metadata since now we expect a different error message if it's empty vs set to something else vs set to our datacenter.

I think it's okay to remove the per-config-entry cases because
1. they were testing the same underlying code
2. I would need to add config entry x cases permutations to test this new functionality which I think is going to make the tests too complicated
3. each config entry has unit tests to ensure it implements the ConfigEntry interface properly (which is all we require to be working here)

Reply:
https://github.com/hashicorp/consul-k8s/pull/424